src/server/cmd/job-shim/main.go

```diff
@@ -30,6 +38,74 @@ func main() {
 	env.Main(do, &appEnv{})
 }
 
+func downloadInput(c *client.APIClient, commitMounts []*fuse.CommitMount) error {
+	for _, commitMount := range commitMounts {
+		if commitMount.Alias == "prev" || commitMount.Alias == "out" {
+			continue
+		}
+		repo := commitMount.Commit.Repo.Name
+		commitID := commitMount.Commit.ID
+		var fromCommitID string
+		if commitMount.DiffMethod != nil && commitMount.DiffMethod.FromCommit != nil {
+			fromCommitID = commitMount.DiffMethod.FromCommit.ID
+		}
+		var fullFile bool
+		if commitMount.DiffMethod != nil {
+			fullFile = commitMount.DiffMethod.FullFile
+		}
+		shard := commitMount.Shard
+		fileInfos, err := c.ListFile(repo, commitID, "/", fromCommitID, fullFile, shard, false)
```

Comment:
Good comments.  I copied the download/upload code from my PoC and forgot that it was just a PoC.